operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -87,9 +98,290 @@ private Deployment fetchExistingDeployment() {
                 .get();
     }
 
+    public void validatePodTemplate() {
```

Comment:
@vmuzikar I think is fair to completely fail if the provided `podTemplate` tries to override important things, wdyt?